libraries/eosiolib/core/eosio/crypto_ext.hpp

```diff
@@ -48,14 +49,183 @@ namespace eosio {
          return dg;
       }
    }
-      
+
    /**
     *  @defgroup crypto Crypto
     *  @ingroup core
     *  @brief Defines API for calculating and checking hashes which
     *  require activating crypto protocol feature
     */
 
+   /**
+    * Abstracts G1 and G2 points. Ensures sizes of x and y are valid
+    *
+    *  @ingroup crypto
+    */
+   struct point_view {
+      /**
+       * Pointer to the x coordinate
+       */
+      char* x;
```

Comment:
Because you need to be able to return a point_view in some of the functions, I suggest that we have point_view and point. Where point holds std::vectors for x and y so we don't need to worry about memory issues.

Reply:
Thanks for your suggestions! I was struggling on how to mutate point_view to return a result.